ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/ScanConfigs/components/ScanConfigClustersTable.tsx

```diff
@@ -0,0 +1,148 @@
+import { TableComposable, Tbody, Td, Th, ThProps, Thead, Tr } from '@patternfly/react-table';
+
+import IconText from 'Components/PatternFly/IconText/IconText';
+import { ClusterScanStatus } from 'services/ComplianceEnhancedService';
+import { getClusterStatusObject } from '../compliance.scanConfigs.utils';
+
+type ScanConfigClustersTableProps = {
+    clusterScanStatuses: ClusterScanStatus[];
+};
+
+function ScanConfigClustersTable({ clusterScanStatuses }: ScanConfigClustersTableProps) {
+    const [page, setPage] = useState(1);
+    const [perPage, setPerPage] = useState(20);
+
+    const onSetPage = (
+        _event: React.MouseEvent | React.KeyboardEvent | MouseEvent,
+        newPage: number
+    ) => {
+        setPage(newPage);
+    };
+
+    const onPerPageSelect = (
+        _event: React.MouseEvent | React.KeyboardEvent | MouseEvent,
+        newPerPage: number,
+        newPage: number
+    ) => {
+        setPerPage(newPerPage);
+        setPage(newPage);
+    };
+
+    // Index of the currently sorted column
+    // Note: if you intend to make columns reorderable, you may instead want to use a non-numeric key
+    // as the identifier of the sorted column. See the "Compound expandable" example.
+    const [activeSortIndex, setActiveSortIndex] = useState<number>(0);
+
+    // Sort direction of the currently sorted column
+    const [activeSortDirection, setActiveSortDirection] = useState<'asc' | 'desc'>('asc');
+
+    // Since OnSort specifies sorted columns by index, we need sortable values for our object by column index.
+    // This example is trivial since our data objects just contain strings, but if the data was more complex
+    // this would be a place to return simplified string or number versions of each column to sort by.
+    const getSortableRowValues = (cluster: ClusterScanStatus): (string | number | null)[] => {
+        // @ts-expect-error api
+        const { clusterName, provider, status } = cluster;
+
+        return [clusterName, null, provider, status] as (string | number | null)[];
+    };
+
+    // Note that we perform the sort as part of the component's render logic and not in onSort.
+    // We shouldn't store the list of data in state because we don't want to have to sync that with props.
+    const sortedClusters = clusterScanStatuses.sort((a, b) => {
+        const aValue = getSortableRowValues(a)[activeSortIndex];
+        const bValue = getSortableRowValues(b)[activeSortIndex];
+        if (typeof aValue === 'number') {
+            // Numeric sort
+            if (activeSortDirection === 'asc') {
+                return aValue - (bValue as number);
+            }
+            return (bValue as number) - aValue;
+        }
+        if (typeof aValue === 'string') {
+            // String sort
+            if (activeSortDirection === 'asc') {
+                return aValue.localeCompare(bValue as string);
+            }
+            return (bValue as string).localeCompare(aValue);
+        }
```

Comment:
Minor, but what do you think about the following to remove the need for `as` here:
```suggestion
        if (typeof aValue === 'number' && typeof bValue === 'number') {
            // Numeric sort
            if (activeSortDirection === 'asc') {
                return aValue - bValue;
            }
            return bValue - aValue;
        }
        if (typeof aValue === 'string' && typeof bValue === 'string') {
            // String sort
            if (activeSortDirection === 'asc') {
                return aValue.localeCompare(bValue);
            }
            return bValue.localeCompare(aValue);
        }
```
Given the logic it shouldn't ever happen, but if somehow `aValue` was a `string` and `bValue` was a `number`, the second case here would throw a runtime error.